sdk/communication/Azure.ResourceManager.Communication/tests/ScenarioTests/SuppressionListTests.cs

```diff
@@ -0,0 +1,286 @@
+                }
+            }
+            Assert.AreEqual(expectedPageCount, pageCount);
+        }
+
+        [Test]
+        public async Task Create_SuppressionListAddresses_Get_SinglePage()
+        {
+            var listName = Recording.Random.NewGuid().ToString();
+            var suppressionList = await CreateDefaultSuppressionListResource(_domainResource, listName);
+            Assert.IsNotNull(suppressionList);
+
+            var addresses = new string[] { "user1@email.com", "user2@email.com", "user3@email.com" };
+            var resourceNames = new Dictionary<string, string>();
+
+            foreach (var address in addresses)
+            {
+                var resource = await CreateDefaultSuppressionListAddressResource(suppressionList, address);
+                resourceNames[resource.Data.Name] = address;
+            }
+
+            int expectedPageCount = 1;
+            int pageCount = 0;
+            var collection = suppressionList.GetSuppressionListAddressResources();
+
+            string continuationToken = default;
+
+            await foreach (var page in collection.GetAllAsync().AsPages(continuationToken: continuationToken))
+            {
+                pageCount++;
+                Assert.AreEqual(resourceNames.Count, page.Values.Count);
+                foreach (var resource in page.Values)
+                {
+                    Assert.AreEqual(resourceNames[resource.Data.Name], resource.Data.Email);
+                }
+                continuationToken = page.ContinuationToken;
+            }
+            Assert.AreEqual(expectedPageCount, pageCount);
+        }
+
+        [Test]
+        public async Task Create_SuppressionListAddresses_Get_MultiplePages()
+        {
+            int pageSize = 10;
+            var listName = Recording.Random.NewGuid().ToString();
+            var suppressionList = await CreateDefaultSuppressionListResource(_domainResource, listName);
+
+            Assert.IsNotNull(suppressionList);
+
+            var resourceNames = new Dictionary<string, string>();
+
+            for (int i = 0; i < 20; i++)
+            {
+                var address = $"{i:00}-{listName}@email.com";
+                var resource = await CreateDefaultSuppressionListAddressResource(suppressionList, address);
+                resourceNames[resource.Data.Name] = address;
+            }
+
+            int expectedPageCount = resourceNames.Count / pageSize;
+            int pageCount = 0;
+            var collection = suppressionList.GetSuppressionListAddressResources();
+
+            string continuationToken = default;
+
+            await foreach (var page in collection.GetAllAsync().AsPages(continuationToken: continuationToken, pageSizeHint: pageSize))
+            {
+                pageCount++;
+
+                // // todo: investigate why pageSize is not working.
+                // Assert.AreEqual(pageSize, page.Values.Count);
+
```

Comment:
Determined the issue, it is in the service side. I will prepare fix @kagbakpem 
When a response has no more pages, I should be returning a null nextLink value. Because the nextLink is never null, the sdk framework will never find an end to a collection when calling GetAllAsync().